### PR TITLE
ISPN-9996 removed clustername field

### PR DESCRIPTION
**Pull request by rigazilla**
This PR removes clustername field form CR Spec. Cluster are identified by metadata.name now.

### CODE REVIEW

File: deploy/cr/cr_minimal.yaml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 apiVersion: infinispan.org/v1
 kind: Infinispan
 metadata:
-  name: example-infinispan
```
Comment (galderz) [inline deploy/cr/cr_minimal.yaml]: Any particular reason to change this name? I'd leave it as is (and also in `cr_with_configmap`). You'd normally run one or the other as you try it out things, so it's fine to have the same example name.

Comment (galderz) [reply deploy/cr/cr_minimal.yaml]: I consider these files as example files, mostly for documentation or our tutorials, so I think it's easier to stick just one name here.

File: test/e2e/main_test.go [hunk 2]
```diff
@@ -172,54 +171,51 @@ func TestCreateWithInternalConfig(t *testing.T) {
 			Kind:       "Infinispan",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name: "cache-infinispan-1",
+			Name: "cache-infinispan-minimal",
```
Comment (galderz) [inline test/e2e/main_test.go]: I think using more descriptive names here is fine.

File: deploy/cr/cr_minimal.yaml (hunk 1 above)
Comment (rigazilla) [reply deploy/cr/cr_minimal.yaml]: I can't remember what was the idea... so I guess it's ok to use just one name

**Comment by rigazilla** [comment]
> On top of comments, I see a few formatting changes here (or change of import orders). Did you add those or did golang's fmt introduce them?

vscode does it automatically with fmt

**Pull request merged**